packages/libs/react-ui/src/components/Button/ButtonIcon.tsx

```diff
@@ -1,11 +1,12 @@
 import { SystemIcon } from '@components/Icon';
 import React, { FC } from 'react';
 
-export interface IButtonIconProps {
+type IconProps = JSX.IntrinsicElements['i'];
+export interface IButtonIconProps extends IconProps {
   icon: (typeof SystemIcon)[keyof typeof SystemIcon];
 }
 
-export const ButtonIcon: FC<IButtonIconProps> = ({ icon }) => {
+export const ButtonIcon: FC<IButtonIconProps> = ({ icon, className }) => {
   const Icon = icon;
-  return <Icon size="md" />;
+  return <Icon size="md" className={className} />;
```

Comment:
What is the reason for passing className to the Icon?

Reply:
Because I pass the `iconClassname` when the button is in a loading state, which animates the loading icon.

Reply:
Would it be possible to apply the animation via the style attribute? I saw it being done in the transfer app like this 
```
<SystemIcon.Loading
                style={{ animation: '2000ms infinite linear spin' }}`
```

This way we don't make the icon component more flexible. 

If not, I think this is fine as well. More of a nitpick